web_responsive/static/src/js/web_responsive.js

```diff
@@ -270,7 +270,12 @@ odoo.define('web_responsive', function (require) {
         _onAppsMenuItemClicked: function (ev) {
             ev.preventDefault();
             ev.stopPropagation();
-            $(ev.currentTarget).find('img').addClass('o-app-icon-waiting');
+            // Prevents anim more app-icon if user click other before action
+            // is fully loaded
+            this.$el.find('.o-app-waiting').removeClass(
+                'o-app-waiting');
+            $(ev.currentTarget).addClass('o-app-waiting');
+            document.body.style.cursor = 'progress';
```

Comment:
Yikes, I have to admit I don't like this approach... A random bug in the JS call stack could leave the cursor in progress mode and confuse the user... As a general rule of thumb, one also try to avoid inline styles as well as `!important` flags in modules. :thinking: 

Yikes, how many rules of thumb I have nowadays! :sweat_smile: 

How about this? You can add a class to the whole app drawer (`this.$el.addClass(".o-app-waiting")`). Then apply the CSS like:

```scss
.o-app-waiting {
    cursor: progress;
    .active img {
        // apply animation here
    }
}
```

Another option would be to just use the waiting overlay that is shipped by odoo  (the famous one of "go grab a coffee"). This would seem like a more integrated UX and probably would mean less code to maintain. Did you consider this option?

Reply:
I don't like the Odoo overlay because I associate it to "long loading" period... see it all the time can give the feeling of a slow system, when it is not.
To avoid apply styles directly i change it to add a class instead: ```$(document.body).addClass('o-cursor-waiting');``` Is applied to whole page because the user can hide the AppDrawer before action is completed.
Now the check to remove the 'waiting class' are done "always" (sucess and fail).